twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelTemplateEditor.tsx

```diff
@@ -25,17 +28,34 @@ const ChannelTemplateEditor: FC<Props> = ({
   const [description, setDescription] = useState<string>(channel?.description || '');
   const [visibility, setVisibility] = useState<string>(channel?.visibility || 'public');
   const [defaultChannel, setDefaultChannel] = useState<boolean>(false);
+  const [group, setGroup] = useState<string>(channel?.channel_group || '');
+  const { companyId, workspaceId } = RouterServices.useStateFromRoute();
+  const url: string = `/channels/v1/companies/${companyId}/workspaces/${workspaceId}/channels/`;
+  const channelsCollection = Collection.get(url, ChannelResource);
+  const channels = channelsCollection.useWatcher({});
 
   useEffect(() => {
     onChange({
       icon,
       name,
       description,
       visibility,
+      channel_group: group,
       default: defaultChannel,
     });
   });
 
+  const getGroups = (channels: ChannelResource[]) => {
+    const groupsNames: string[] = [];
+    channels
+      .sort((a, b) => (a.data.channel_group || '').localeCompare(b.data.channel_group || ''))
+      .map((channel: ChannelResource) => {
```

Comment:
`map` is for transforming the input to output. Here you can use a `forEach` or use `map` like:

```js
const grousNames: string[] = channels.sort(sortFn).map(mapFn).filter(Boolean);
```

Reply:
Updated 👍🏻